src/plugins/gs-flatpak.c

```diff
@@ -2039,3 +2126,10 @@ gs_flatpak_app_is_runtime (GsApp *app)
 {
 	return g_strcmp0 (gs_app_get_flatpak_kind_as_str (app), "runtime") == 0;
 }
+
+void
+gs_flatpak_set_download_updates (GsFlatpak *self,
+				 gboolean download_updates)
+{
+	self->download_updates = download_updates;
```

Comment:
No refresh needs to happen IMO (as it won't really compromise anything: we always check the flag directly before performing ops). In any case a refresh would have to be done from outside this module.